internal/crypto/passwap.go

```diff
@@ -0,0 +1,175 @@
+package crypto
+
+import (
+	"fmt"
+
+	"github.com/mitchellh/mapstructure"
+	"github.com/zitadel/passwap"
+	"github.com/zitadel/passwap/argon2"
+	"github.com/zitadel/passwap/bcrypt"
+	"github.com/zitadel/passwap/md5"
+	"github.com/zitadel/passwap/scrypt"
+	"github.com/zitadel/passwap/verifier"
+
+	"github.com/zitadel/zitadel/internal/errors"
+)
+
+type HashName string
+
+const (
+	HashNameArgon2   HashName = "argon2"   // used for the common argon2 verifier
+	HashNameArgon2i  HashName = "argon2i"  // hash only
+	HashNameArgon2id HashName = "argon2id" // hash only
+	HashNameBcrypt   HashName = "bcrypt"   // hash and verify
+	HashNameMd5      HashName = "md5"      // verify only, as hashing with md5 is insecure and deprecated
+	HashNameScrypt   HashName = "scrypt"   // hash and verify
+)
+
+type PasswordHashConfig struct {
+	Verifiers []HashName
+	Hasher    HasherConfig
```

Comment:
we can also un-nest this if that would look better in the config.

Reply:
for me it's ok like that